source/adios2/common/ADIOSConfig.h.in

```diff
@@ -25,8 +25,16 @@
  *     Id: @CMAKE_Fortran_COMPILER_ID@ @CMAKE_CXX_COMPILER_VERSION@
  */
 
-/* Everything past this line is programatically generated */
+/* Everything between here and the note below is programatically generated */
 
 @ADIOS2_CONFIG_DEFINES@
 
+/* Everything between here and the note above is programatically generated */
+
+#ifdef ADIOS2_HAVE_MPI
+#  define ADIOS2_USE_MPI 1
+#else
+#  define ADIOS2_USE_MPI 0
+#endif
+
```

Comment:
I think this needs to be more complex to allow the user to appropriately override it.  Something like:
```c++
#ifndef ADIOS2_USE_MPI
  #ifdef ADIOS2_HAVE_MPI
    #define ADIOS2_USE_MPI 1
  #else
    #define ADIOS2_USE_MPI 0
  #endif
#else
  #if ADIOS2_USE_MPI && !(defined ADIOS2_HAVE_MPI)
    #error "ADIOS was not built with MPI support"
  #endif
#endif
```

Reply:
As of this PR we don't allow the application to pick yet.  That logic will be updated in the main transition PR.

Reply:
To clarify: without the library split the application cannot pick.  The ability to pick will be added in the main library split PR.